### PR TITLE
8304053: Revert os specific stubs for SystemMetrics

### PR DESCRIPTION
**Pull request by adamfarley**
This change excludes the duplicated class SystemMetrics, and ensures we use the platform-specific version of the SystemMetrics class instead.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304053](https://bugs.openjdk.org/browse/JDK-8304053): Revert os specific stubs for SystemMetrics

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)
 * [Andrew John Hughes](https://openjdk.org/census#andrew) (@gnu-andrew - **Reviewer**) ⚠️ Review applies to [b6a882fe](https://git.openjdk.org/jdk8u/pull/40/files/b6a882fe357ce295d35514ce0068f6f1785f6f66)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u pull/40/head:pull/40` \
`$ git checkout pull/40`

Update a local copy of the PR: \
`$ git checkout pull/40` \
`$ git pull https://git.openjdk.org/jdk8u pull/40/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 40`

View PR using the GUI difftool: \
`$ git pr show -t 40`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u/pull/40.diff">https://git.openjdk.org/jdk8u/pull/40.diff</a>

</details>

### CODE REVIEW

**Comment by jerboaa** [comment]
@adamfarley Please enable GHA.

**Comment by adamfarley** [comment]
I thought I did.

**Comment by adamfarley** [comment]
It seems to be running something. Is this not GHA?

https://github.com/adamfarley/jdk8u/actions/runs/4417066915

**Comment by jerboaa** [comment]
> It seems to be running something. Is this not GHA?
> 
> https://github.com/adamfarley/jdk8u/actions/runs/4417066915

Yes, that's GHA. I mentioned it, because #37 didn't have it enabled causing the breakage :-(

**Comment by adamfarley** [comment]
> > It seems to be running something. Is this not GHA?
> > https://github.com/adamfarley/jdk8u/actions/runs/4417066915
> 
> Yes, that's GHA. I mentioned it, because #37 didn't have it enabled causing the breakage :-(

Fair enough. Better to be safe than sorry. Again. :(

**Review by jerboaa — CHANGES_REQUESTED** [review]
We should only exclude the class on Linux. That's the OS where the `SystemMetrics` class is actually implemented. In order to solve the resulting build break on Mac and AIX, we should revert additions of the `null` returning classes there. Perhaps with #39.

I propose the following course of action to fix it:

1. Only exclude `SystemMetrics.java` on Linux from the `solaris` directory in this PR. Make this PR dependent on #39 (dependent pull request).
2. Revert stub impl additions on Mac OS X and AIX with #39.

@gnu-andrew  Thoughts?

**Comment by jerboaa** [comment]
> We should only exclude the class on Linux. That's the OS where the `SystemMetrics` class is actually implemented. In order to solve the resulting build break on Mac and AIX, we should revert additions of the `null` returning classes there. Perhaps with #39.
> 
> I propose the following course of action to fix it:
> 
>     1. Only exclude `SystemMetrics.java` on Linux from the `solaris` directory in this PR. Make this PR dependent on #39 (dependent pull request).
>     2. Revert stub impl additions on Mac OS X and AIX with #39.

Change of plan. Do all of it here, and use JDK-8304053 as the bug to integrate.

**Comment by gnu-andrew** [comment]
Ok, this just changed while I was looking at it :)

I think the current version incorporates #39 now, so we just need this one?

This seems to relate to what I was referring to in [my comment](https://bugs.openjdk.org/browse/JDK-8303828?focusedCommentId=14565610&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14565610) on the bug about 11u. 11u uses `src/unix` for the generic versions and platform-specific overrides. On 8u, `src/solaris` is that generic version unless we opt out.

So I would also remove the unused `unix` copy of the file here, assuming builds aren't broken, and refer to both bug IDs.

**Comment by adamfarley** [comment]
Sure thing. I've changed the PR title as requested, and will remove the unix stub class. Will not squash or force push.

File: jdk/make/CompileJavaClasses.gmk [hunk 1]
```diff
@@ -307,6 +307,7 @@ endif
 LINUX_SRC_DIRS :=
 ifeq ($(OPENJDK_TARGET_OS),linux)
      LINUX_SRC_DIRS += $(JDK_TOPDIR)/src/linux/classes
+     EXFILES += $(JDK_TOPDIR)/src/solaris/classes/jdk/internal/platform/SystemMetrics.java
```
[the remark this answers is not in the document]
Comment (adamfarley) [reply jdk/make/CompileJavaClasses.gmk:310]: Sure. 1 moment.

[the remark this answers is not in the document]
Comment (adamfarley) [reply jdk/make/CompileJavaClasses.gmk:313]: Done.

[the remark this answers is not in the document]
Comment (jerboaa) [reply jdk/make/CompileJavaClasses.gmk:313]: Thanks!

**Review by jerboaa — APPROVED** [review]
Looks good. Let's wait for a clean GHA run this time before we integrate :)

**Comment by gnu-andrew** [comment]
FYI: Linux builds are broken without this too:

~~~
/home/andrew/projects/openjdk/upstream/jdk8u-dev/jdk/src/linux/classes/jdk/internal/platform/SystemMetrics.java:27: error: duplicate class: jdk.internal.platform.SystemMetrics
public class SystemMetrics {
~~~

so I'm not going to promote b06 from last week, but will instead make b06 the promotion at the end of this week, with this fix. I'll write similar to the mailing list.

**Comment by jerboaa** [comment]
@adamfarley The bug is approved. GHA looks OK. Please `/integrate`. Thanks!

**Pull request closed without merging**